scripts/irods/test/test_setting_user_password.py

```diff
@@ -0,0 +1,1088 @@
+	@unittest.skipIf(test.settings.TOPOLOGY_FROM_RESOURCE_SERVER, "Must configure catalog provider for this test.")
+	def test_invalid_password_storage_mode_value_causes_configuration_reload_to_fail(self):
+		# A test to see what the server does when the configuration is an invalid value cannot be created because the
+		# server validates the configuration before starting. The invalid configuration causes the validation to fail
+		# and so the configuration does not reload. If we could test this, we would expect setting user passwords to
+		# fail for all cases and return an error because the configuration cannot direct what should happen.
+		config = IrodsConfig()
+		with lib.file_backed_up(config.server_config_path):
+			# Update server config and reload configuration. This should cause an error.
+			config.server_config["user_password_storage_mode"] = "banana"
+			lib.update_json_file_from_dict(config.server_config_path, config.server_config)
+			with self.assertRaises(IrodsError):
+				IrodsController(config).reload_configuration(max_retries=3)
+
+		IrodsController().reload_configuration()
+
+	def test_igroupadmin_mkuser_no_scramble_option_requires_optional_zone_to_be_specified(self):
+		user_name = "nobody"
+		with session.make_session_for_existing_admin() as admin_session:
+			admin_session.assert_icommand(
+				["igroupadmin", "mkuser", user_name, "whatever", "no-scramble"],
+				"STDERR", "-828000 CAT_INVALID_ZONE")
+
+			self.assertFalse(lib.user_exists(admin_session, user_name))
+
+
+class ipasswd_test_base(unittest.TestCase):
+	@classmethod
+	def setUpClass(self):
+		# Create an admin user / session for all the tests so we don't have to keep making it for each test.
+		self.admin = session.mkuser_and_return_session("rodsadmin", "otherrods", "rods", lib.get_hostname())
+
+		self.server_config_backup = tempfile.NamedTemporaryFile(prefix=os.path.basename(paths.server_config_path())).name
+		shutil.copyfile(paths.server_config_path(), self.server_config_backup)
+		with open(paths.server_config_path()) as f:
+			server_config = json.load(f)
+		server_config["user_password_storage_mode"] = self.password_storage_mode
+		# Configure insecure mode for irods authentication scheme if the tests are not being run with TLS enabled.
+		if False == test.settings.USE_SSL:
+			server_config['plugin_configuration']['authentication']['irods'] = {'insecure_mode': True}
+		# Update the server configuration and reload the configuration.
+		lib.update_json_file_from_dict(paths.server_config_path(), server_config)
+		IrodsController().reload_configuration()
+
+		self.old_password = "old_password"
+		self.new_password = "new_password"
+		self.ipasswd_input = "{}\n{}\n{}\n".format(self.old_password, self.new_password, self.new_password)
+
+	@classmethod
+	def tearDownClass(self):
+		# Restore the original server configuration.
+		shutil.copyfile(self.server_config_backup, paths.server_config_path())
+		IrodsController().reload_configuration()
```

Comment:
I noticed that other tests wrap this in a conditional which checks the `USE_SSL` option.

Why doesn't this follow the same pattern?